### PR TITLE
feat(refs DPLAN-11981): Add image to modal

### PR DESCRIPTION
**Pull request by gruenbergerdemos**
Ticket: https://demoseurope.youtrack.cloud/issue/DPLAN-11981

The user should be able to view images enlarged. 
- DpEditor: On click+strg the image opens in the DpUploadModal where the alt text can be edited, too. 
- DpUploadModal: Make button text dynamically
- Fix bug in DpResizableImage: Observer should be disconnected before the component is destroyed
- Fix bug in DpEditor: Open image modal on click only if image is child of editor
- Fix bug in DpUploadModal: Reset content after closing (clicking on x)

### CODE REVIEW

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Since v0.0.10, this Changelog is formatted according to the [Common Changelog][c
 ### Fixed
 
 - ([#917](https://github.com/demos-europe/demosplan-ui/pull/917)) Fix cut methode for TipTap Editor. ([@ahmad-demos](https://github.com/ahmad-demos))
+- ([#919](https://github.com/demos-europe/demosplan-ui/pull/919)) DpResizableImage: Disconnect image observer before destroying component ([@gruenbergerdemos](https://github.com/gruenbergerdemos))
```
[the remark this answers is not in the document]
Comment (gruenbergerdemos) [reply CHANGELOG.md:10]: Yes, the text is different. One is a bug fix, the other an added feature. Do you think it is confusing?

[the remark this answers is not in the document]
Comment (salisdemos) [reply CHANGELOG.md:10]: Yes a bit. But ok. It won't hurt.

File: src/components/DpDataTable/DpDataTable.vue [hunk 2]
```diff
@@ -44,6 +44,13 @@
       <tbody
         v-if="!isDraggable && !isLoading"
         :data-cy="`${dataCy}:tbody`">
+      <dp-modal
+          ref="imgModal"
+          content-classes="w-fit">
```
[the remark this answers is not in the document]
Comment (gruenbergerdemos) [reply src/components/DpDataTable/DpDataTable.vue]: Done in [ec86609](https://github.com/demos-europe/demosplan-ui/pull/919/commits/ec866092d4df684befd4e55e290da29a099d0424)

File: src/components/DpEditor/DpEditor.vue [hunk 3]
```diff
@@ -1089,9 +1098,12 @@ export default {
...
     })
 
-    this.$root.$on('open-image-alt-modal', (e, id) => {
-      this.editingImage = id
-      this.openUploadModal({ editAltOnly: true, currentAlt: e.target.getAttribute('alt') })
+    this.$root.$on('open-image-alt-modal', ({ event, imgId, editorId }) => {
+      if (this.editor.options.id !== editorId) {
+        return
+      }
+      this.editingImage = imgId
```
Comment (salisdemos) [inline src/components/DpEditor/DpEditor.vue:1105]:
```suggestion
      }

      this.editingImage = imgId
```

**Review by salisdemos — CHANGES_REQUESTED** [review]
just nitpics

File: src/components/DpEditor/DpUploadModal.vue [hunk 4]
```diff
@@ -107,21 +119,30 @@ export default {
...
-      this.resetAndClose()
+      this.reset()
+      this.closeModal()
     },
 
-    resetAndClose () {
+    handleAbort () {
+      this.reset()
+      this.closeModal()
+    },
```
[the remark this answers is not in the document]
Comment (gruenbergerdemos) [reply src/components/DpEditor/DpUploadModal.vue]: Done with [8ae885d](https://github.com/demos-europe/demosplan-ui/pull/919/commits/8ae885d1214d29ebe7989e7322e87ee135191721)

File: src/components/DpEditor/DpUploadModal.vue [hunk 5]
```diff
@@ -107,21 +119,25 @@ export default {
...
+
     emitAndClose () {
       if (this.editAltTextOnly) {
         this.$emit('add-alt', this.altText)
       } else if (this.fileUrl) {
         this.$emit('insert-image', this.fileUrl, this.altText)
       }
-      this.resetAndClose()
+      this.closeAndReset()
```
Comment (salisdemos) [inline src/components/DpEditor/DpUploadModal.vue:134]:
```suggestion

      this.closeAndReset()
```

**Review by salisdemos — APPROVED** [review]

**Pull request merged**